drivers/gnss/u_blox_protocol/u_blox_protocol.c

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2024 NXP
+ * Copyright (c) 2022 Abel Sensors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "u_blox_protocol.h"
+
+const uint32_t u_blox_baudrate[U_BLOX_BAUDRATE_COUNT] = {
+	4800,
+	9600,
+	19200,
+	38400,
+	57600,
+	115200,
+	230400,
+	460800,
+};
+
+static void u_blox_create_frame(uint8_t ubx_frame[], uint16_t *ubx_frame_size,
```

Comment:
please use `uint8_t *ubx_frame` instead of `ubx_frame[]`, the latter should only be used for fixed size arrays.